Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-contests.tsx

```diff
@@ -95,24 +116,44 @@ const ContestsProvider = ({ children }: IContestsProviderProps) => {
         () => strategiesAreLoaded && categoriesAreLoaded
             ? generateStatusFilters()
                 .concat(generateCategoryFilters(categories))
-                .concat(generateStrategyFilters(strategies))
-            : [],
+                .concat(generateStrategyFilters(strategies)) as IFilter[]
+            : [] as IFilter[],
         [ categories, categoriesAreLoaded, strategies, strategiesAreLoaded ],
     );
 
+    const possibleSortingTypes = useMemo(
+        () => generateSortingStrategy() as unknown as ISort[],
+        [],
+    );
+    
     const filters = useMemo(
-        () => collectFilters(params, possibleFilters),
+        () => collectParams(params, possibleFilters, DEFAULT_FILTER_TYPE, DEFAULT_STATUS_FILTER_TYPE),
         [ params, possibleFilters ],
     );
 
+    const sortingTypes = useMemo(
+        () => collectParams(params, possibleSortingTypes, DEFAULT_SORT_FILTER_TYPE, DEFAULT_SORT_TYPE),
+        [ params, possibleSortingTypes ],
+    );
+
     const currentPage = useMemo(
         () => collectCurrentPage(params),
         [ params ],
     );
 
     const clearFilters = useCallback(
-        () => clearParams(),
-        [ clearParams ],
+        (type: FilterType, defaultId: number) => {
+            if (type === FilterType.Sort) {
+                unsetParam(type);
+                setParam(type, defaultId);
+            } else {
+                unsetParam(FilterType.Status);
```

Comment:
Maybe put these in an array filterTypes and use it here?

Reply:
I made the clear filters function to accept an array of filter types and if we pass both sorting and some filter type to it, it will clear the filters and the sorting simultaneously.